tools/protodoc/protodoc.py

```diff
@@ -57,12 +57,31 @@
 This extension may be referenced by the qualified name *$extension*
 
 .. note::
+  EXTENSION_MARKER
+
   $status
 
   $security_posture
 
+  This extension extends the following extension type:
+
+$categories
+
 """)
 
+
+EXTENSION_TYPE_TEMPLATE = string.Template("""$anchor
+
+.. tip::
+  EXTENSION_TYPE_MARKER
+
+  This extension type has the following known extensions
```

Comment:
im a bit concerned about nomenclature - ie extension, type, category etc it would be good to fix on a particular terminology

Reply:
Extension is an extension, e.g. the HTTP buffer filter.

I think we should use category to refer to "HTTP filter". Drop the idea of "extension type" across the board.

Reply:
updated `type` -> `category` across PR